Corgibytes.Freshli.Cli/CommandRunners/Git/CheckoutHistoryCommandRunner.cs

```diff
@@ -1,18 +1,32 @@
 using System;
 using System.CommandLine.Invocation;
+using System.CommandLine.IO;
 using Corgibytes.Freshli.Cli.CommandOptions;
+using Corgibytes.Freshli.Cli.Functionality.Git;
 using Corgibytes.Freshli.Lib;
 
 namespace Corgibytes.Freshli.Cli.CommandRunners.Git;
 
 public class CheckoutHistoryCommandRunner : CommandRunner<CheckoutHistoryCommandOptions>
 {
-    public CheckoutHistoryCommandRunner(IServiceProvider serviceProvider, Runner runner) : base(serviceProvider, runner)
+    private readonly GitArchive _gitArchive;
+
+    public CheckoutHistoryCommandRunner(IServiceProvider serviceProvider, Runner runner, GitArchive gitArchive) : base(serviceProvider, runner)
     {
+        _gitArchive = gitArchive;
```

Comment:
I'm trying to understand this part. We are passing a `gitArchive`, and then populating it with data in `Run`. Is there a particular reason we don't just create a new `GitArchive` object here? (There may well be, I'm just looking for more insight.)

Reply:
Good point. I think I've made a brain fart. My intent is to separate the thing that creates the Git Archive from the command runner but we don't need an extra object for that.

Reply:
So I was looking at it, not quite figuring out why I did that but now I know.
It's indeed following the builder pattern. I don't want the way to create a GitArchive, or at least, the way of gathering all the necessary information tied to an object specific to the command line (`CheckoutHistoryCommandRunner`). If we decide to switch the user interface or the architecture where we don't use the CommandRunner as outward facing, then the refactor (in this instance) becomes more of a hassle. Same goes if we want to add an interface, or add architecture next to command runners, we need to think about what data to collect to create an archive. 

I'd agree that in this instance the services needed to create an archive are minimal but I find it good practice nonetheless.

Reply:
It is the Builder pattern, but it's value is a bit lost. 
There's two objects in play here `GitArchive` and `GitArchiveProcess`. I separated the two because I wanted to unit test the `GitArchive`. It has a test, but that's testing it returns a string. 

I'm going to merge the two. :-) Thanks for pointing it out